addons/stock/models/stock_move.py

```diff
@@ -887,6 +888,10 @@ def _update_reserved_quantity(self, need, available_quantity, location_id, lot_i
             to_update = self.move_line_ids.filtered(lambda m: m.product_id.tracking != 'serial' and
                                                     m.location_id.id == reserved_quant.location_id.id and m.lot_id.id == reserved_quant.lot_id.id and m.package_id.id == reserved_quant.package_id.id and m.owner_id.id == reserved_quant.owner_id.id)
             if to_update:
+                uom_quantity = self.product_id.uom_id._compute_quantity(quantity, to_update.product_uom_id, rounding_method='HALF-UP')
+                uom_quantity = float_round(uom_quantity, precision_digits=rounding)
+                uom_quantity_back_to_product_uom = to_update.product_uom_id._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
+            if to_update and float_compare(quantity, uom_quantity_back_to_product_uom, precision_digits=rounding) == 0:
                 to_update[0].with_context(bypass_reservation_update=True).product_uom_qty += self.product_id.uom_id._compute_quantity(quantity, to_update[0].product_uom_id, rounding_method='HALF-UP')
```

Comment:
I would write it like this : 
```python
 if to_update:
    uom_quantity = self.product_id.uom_id._compute_quantity(quantity, to_update.product_uom_id, rounding_method='HALF-UP')
    uom_quantity = float_round(uom_quantity, precision_digits=rounding)
    uom_quantity_back_to_product_uom = to_update.product_uom_id._compute_quantity(uom_quantity, self.product_id.uom_id, rounding_method='HALF-UP')
    if float_compare(quantity, uom_quantity_back_to_product_uom, precision_digits=rounding) == 0:
        to_update[0].with_context(bypass_reservation_update=True).product_uom_qty += self.product_id.uom_id._compute_quantity(quantity, to_update[0].product_uom_id, rounding_method='HALF-UP')
```
to avoid testing `to_update` two times.
Also this last line can be replaced by
```python
to_update[0].with_context(bypass_reservation_update=True).product_uom_qty += uom_quantity
```
to not convert `quantity` two times

Reply:
Thanks @whenrow, indeed I forgot to add [0] for to_update[0], and there are unnecessary repetitions.

However, for your remark about testing to_update two times, I think we want to create a new move line if quantity <> uom_quantity_back_to_product_uom (quickly checked with amoyaux), go in the "else" create(self._prepare_move_line_vals[...] part, so I kept

`if to_update and float_compare(quantity, uom_quantity_back_to_product_uom, precision_digits=rounding) == 0:`